app/frontend/models/permit-application.ts

```diff
@@ -105,6 +105,16 @@ export const PermitApplicationModel = types
       }
       return response
     }),
+
+    submit: flow(function* (params) {
+      const response = yield self.environment.api.submitPermitApplication(self.id, params)
+      if (response.ok) {
+        const { data: permitApplication } = response.data
+        self.rootStore.permitApplicationStore.mergeUpdate(permitApplication, "permitApplicationMap")
+      }
+      return response
```

Comment:
I think this should be `response.ok`, otherwise the code below will always appear to be successful as the response object is always defined
`const onSubmit = async (submission: any) => {
    if (await permitApplication.submit({ submissionData: submission })) {
      navigate("/permit-applications/sucessful-submission")
    }
  }`

Reply:
good catch!